### PR TITLE
Webp thumbnails

### PR DESCRIPTION
**Pull request by ecto**
Also refactors ThumbFile away from getter/setter pattern

Working on tests still

### CODE REVIEW

File: uploadedfile/thumbfile.go [hunk 1]
```diff
@@ -17,172 +17,94 @@ var (
...
-}
-
 func (this *ThumbFile) SetPath(path string) error {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		return errors.New(fmt.Sprintf("Error when creating thumbnail %s", this.GetName()))
+		return errors.New(fmt.Sprintf("Error when creating thumbnail %s", this.Name))
 	}
 
-	this.localPath = path
+	this.LocalPath = path
```
Comment (briankassouf) [inline uploadedfile/thumbfile.go]: Perhaps `LocalPath` shouldn't be exported since we always want to run it this existence check

Comment (ecto) [inline uploadedfile/thumbfile.go]: Okay, makes sense in the context of `StorableObject`

**Comment by briankassouf** [comment]
this block: https://github.com/Imgur/mandible/pull/79/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR475

requires a dimensions param to be passed in. We should remove this restriction so that we can convert full sized images to webps

**Comment by ecto** [comment]
our travis config doesn't use docker, so it doesn't contain the new webp library needed for the tests. will address this in the morning

**Comment by ecto** [comment]
travis now uses docker

**Comment by ecto** [comment]
actually.. this breaks max_width without specifying width. going to rework validation logic

File: server/server_test.go [hunk 2]
```diff
@@ -243,3 +239,318 @@ func TestAuthentication(t *testing.T) {
...
...6cf/+AHPzy7tzfe3l1kK8csBIVaaUL0DokkERA5togkhEAihp53KQAd9LN+xUyISrKSDokQADyzI2Zki0zgFTAiCgaLIDyDtQxIQRTrKAUUxjoPYNFZ7imxX40B8NOrGboHPlb+jxW5y2X+V3/9N6Px6PyZMxhE6+l+UdTO+BYbdIa9CwlUEPdLbMY5JJBaCkUYSBAI3jOzB2QiloRKsZKAwCRJShQSiQDhXv46O++tA/Z9bBpIKh0myUBqnee5ZWvRMSI4VD1F6eFCEZ+7AfC+jS+8s6puktEIEcuqsG2jGYZhrCUBWCQWUgopSSBJIbVUUSAUoURUAmQvwdmBt4IIpcA4xDBgkigUUYCyF8ruNa8IPKMxtm2sM23XtbbrcxgYoCyK1Wrdmc4L7qesgdZfoQviL7c3fv7HL1y7dm0jUtRUNltPBkMt2RoCYBJCCIzjVIYhMAKgc9azkxIQBLBDdic5nKRACO5RvCQB+zcAAo8Ajq1DzwAeEbwz3lshUGslGFKbJGlChK5X8wABHhm8/H9/6e91tpzOXp3O+u1Lk+T86d3xOAVHcRS7zvjAe+vBVgiIIJy1TB6RwQuwHbLv57zoHVju0xMAyDP2OtnO9oc8eg8MhADghETJhIQMoJRK00EQBFm5FijZAyEJlF+JC/rq27wysF5ub08GaaSlFiQJwRnjjQHrvLXOWuqfrs6BM+AtIGAvGiMEKo2kvEdkQShBSGYE9mgtsgdrXVtb0zpgkhKk8MBMaI3Nsywvcs/cA3WVFF9bA+D/+nyep2aAG4tMNevJ5piUlkqzZ++sd56dt8a41rC13hlkT+C96by1JCToEEgCETCCJwEChGKi/u2PgMF737XWGMC+9EEyCsfsHXjni6I4Opo3TY0MApWSUnz9rv79xCf8Ql3TrYOlyZfp5maUpp0xHsFY33XWe++8c77rTOus8+CbtjGdEUL3KZRI1HseRDqp+3CGnAfnXWds03rvBEkUghGZyDG0xrS1KYv6aD6vm4pIBSpgdvLcuXPHH3G+vh6NAPwXc15CQHdCRvA9Sf2Va/M2/XnGbu+R3a1umARBAJiEYahDLRHAk22kE8hMAECdskihVjIAQiCPxGA6sACEIBQbBMdEUpAESUiCmZnZgkTuAESSjKJ4UBzcAW46bdI0/W8DRsjA5nZi/wAAAABJRU5ErkJggg=="
```
Comment (briankassouf) [inline server/server_test.go]: Can we get a smaller image here?

File: uploadedfile/thumbfile.go [hunk 3]
```diff
@@ -278,18 +229,40 @@ func (this *ThumbFile) processThumb(original *UploadedFile) error {
...
 		}
 	}
 
-	filename, err := processorcommand.CustomThumb(original.GetPath(), this.GetName(), this.ComputeWidth(original), this.ComputeHeight(original), this.GetCropGravity(), cropWidth, cropHeight, this.GetQuality(), thumbType.FromMime(original.GetMime()))
+	width := this.ComputeWidth(original)
+	height := this.ComputeHeight(original)
+	if width == 0 || width > maxImageSideSize {
+		return errors.New("Invalid width")
+	}
+	if height == 0 || width > maxImageSideSize {
```
Comment (briankassouf) [inline uploadedfile/thumbfile.go]: `height`

Comment (ecto) [inline uploadedfile/thumbfile.go]: Already fixed, haven't pushed

File: server/server_test.go (hunk 2 above)
Comment (ecto) [inline server/server_test.go]: It's actually pretty small, I'll shrink it though

Comment (pastudan) [inline server/server_test.go]: ahh, don't shrink me!

Comment (briankassouf) [inline server/server_test.go]: igcvg3pl-gwl

File: uploadedfile/thumbfile.go (hunk 3 above)
Comment (briankassouf) [inline uploadedfile/thumbfile.go]: :+1:

File: server/server.go [hunk 4]
```diff
@@ -460,43 +461,45 @@ func parseThumbs(r *http.Request) ([]*uploadedfile.ThumbFile, error) {
 		return []*uploadedfile.ThumbFile{}, nil
 	}
 
-	var t map[string]map[string]interface{}
-	err := json.Unmarshal([]byte(thumbString), &t)
+	type ThumbRequest struct {
```
Comment (jacobgreenleaf) [inline server/server.go]: :+1:

**Comment by jacobgreenleaf** [comment]
:+1:

**Pull request merged**